event/eth_sign_redeem.go

```diff
@@ -97,60 +97,74 @@ func (j *JobETHSignRedeem) DoMyJob(ctx interface{}) {
 	}
 
 	addr := ethCtx.GetValidatorETHAddress()
+	//Get receipt first ,then status [ other way around might cause ambiguity ]
+	//Confirm redeem tx has been sent by user
 	txReceipt, err := cd.VerifyReceipt(tx.Hash())
 	if err != nil {
 		ethCtx.Logger.Debug("Trying to confirm RedeemTX sent by User Receipt :", err)
 		return
 	}
-	//Failed to delete old version of chainstate err version does not exist version: -900
-	// Get receipt first ,then status [ other way around might cause ambiguity ]
-	// If expired fail tracker
+	/*
+			Points to note
 
-	// If Success is true and validator has send signature implies , validator has signed , but his sign got reverted
-	//     as this was the fourth sign . Retrycount > 0 means that the validator did sign , status was 0 before ,implies
-	//     this is not an old redeem
+		    If expired fail tracker
 
-	// Status is not 0 , and txreceipt is true , redeem tx present in Ethereum ,but redeem status is not ongoing .Fail tracker
+			If Success is true and validator has send signature implies , validator has signed , but his sign got reverted
+			as this was the fourth sign .
 
-	// HasValidatorsigned returns success , means signature confirmed
+		    Retrycount > 0 means that the validator did sign , status was 0 before ,implies this is not an old redeem
 
+			Status is not 0 , and txreceipt is true , redeem tx present in Ethereum ,but redeem status is not ongoing Fail tracker .
+
+			HasValidatorsigned returns success , means signature confirmed
+	*/
+
+	//Checking for confirmation of Vote
 	success, err := cd.HasValidatorSigned(addr, msg.From())
 	if err != nil {
 		ethCtx.Logger.Error("Error connecting to HasValidatorSigned function in Smart Contract  :", j.GetJobID(), err)
 	}
-	status, err := cd.VerifyRedeem(addr, msg.From())
-	if err != nil {
-		ethCtx.Logger.Error("Error in verifying redeem :", j.GetJobID(), err)
-		return
-	}
+	//Signature confirmed
 	if success {
 		ethCtx.Logger.Debug("validator Sign Confirmed | validator Address (SIGNER):", ethCtx.GetValidatorETHAddress().Hex(), "| User Eth Address :", msg.From().Hex())
 		j.Status = jobs.Completed
 		return
 	}
+	//Log print debugger
 	if j.RetryCount >= 0 && !success {
```

Comment:
!success will always be true here.

Reply:
@pav-kap I think what you meat was that it's pointless to check the flag success when error is not nil, right?

Reply:
@pav-kap if that's the case, we can make the above err check as 
if err != nil or !success {
    ethCtx.Logger.Error("some error msg")
}

Reply:
I mean if success is true, it should have already went inside if success {} block, and returned.. Only since its not success, it came down.. So !success is always true here

Reply:
Removed the extra flag